EIPS/eip-7999.md

```diff
@@ -0,0 +1,501 @@
+
+## Motivation
+
+A multidimensional fee market enables precise control over resource consumption. It allows *the market* to fairly price resources according to targets and limits deemed safe by developers, and it allows resources to be consumed at maximum capacity within these limits. Directly expanding the current fee market design to the multidimensional setting can however have negative effects on the user experience (UX) and on economic efficiency. Users are forced to set a `max_fee_per_gas` for each resource, where a too low allocation in any dimension can render the transaction ineligible.
+
+This EIP leverages the natural fungibility of the user's fee budget by letting users set a single unified `max_fee`. Instead of partitioning funds into non-fungible buckets, a single pool of ETH can then be allocated dynamically to cover costs wherever they arise, ensuring a more efficient use of capital. Users will be able to specify a lower `max_fee` than the implied aggregate maximum unless all base fees are perfectly correlated (which reduces to the current design), because they do not need to buffer for spurious price movements in a single resource dimension.
+
+Ethereum's current fee market has "tech debt" in that two separate mechanisms are used—one for regular gas ([EIP-1559](./eip-1559.md)) and one for blob gas ([EIP-4844](./eip-4844.md)). The proposal unifies the fee market under the preferred EIP-4844 design. That design allows for exact control over long-run resource consumption. In a multidimensional setting with individual base fees, we can then for example achieve precise control over state growth, while accommodating temporary spikes. Excess gas of EIP-4844 is further normalized relative to the limit, allowing for a single update fraction across resources that retains current percentage ranges while keeping the price stable if any gas limit changes. 
+
+Calldata is added first, to speed up worst-case payload propagation and expand available EVM gas—without compromising gas introspection. A method for facilitating gas aggregation across resources within the EVM is outlined as an avenue for preserving backward compatibility when expanding further. The logic of EIP-7918 is integrated into the multidimensional setting to ensure that calldata has a higher cost per byte than blob data. 
+
+## Specification
+
+The specification inherits its logic from [EIP-7706](./eip-7706.md), incorporating the changes necessary for facilitating one aggregate fee, a multidimensional [EIP-7918](./eip-7918.md) logic, a systematic approach to [EIP-7805](./eip-7805.md), and a stable gas normalization function, etc.
+
+### Parameters
+
+| Constant | Value | Description |
+| :--- | :--- | :--- |
+| `FORK_BLKNUM` | `TBD` | Block number of fork activation |
```

Comment:
remove fork activation block, just say when the EIP is activated via a hardfork...